nvdaHelper/espeak/speechPlayer.cpp

```diff
@@ -0,0 +1,137 @@
+#include <espeak-ng/espeak_ng.h>
+#include <espeak-ng/speak_lib.h>
+#include "speech.h"
+#include "klatt.h"
+#include "phoneme.h"
+#include "synthesize.h"
+#include "voice.h"
+#include <speechPlayer/src/speechPlayer.h>
+
+extern "C" WGEN_DATA wdata;
+extern "C" voice_t *wvoice;
+extern unsigned char *out_ptr;
+extern unsigned char *out_end;
+
+speechPlayer_handle_t speechPlayerHandle=NULL;
+#define minFadeLength 110
+
+inline bool needsMixWaveFile() {
+	return wdata.n_mix_wavefile>0;
+}
+
+unsigned int mixWaveFile(unsigned int maxNumSamples, sample* sampleBuf) {
+	unsigned int i=0;
+	for(;wdata.mix_wavefile_ix<wdata.n_mix_wavefile;++wdata.mix_wavefile_ix) {
+		if(i>=maxNumSamples) break;
+		int val;
+		if(wdata.mix_wave_scale==0) {
+			val=wdata.mix_wavefile[wdata.mix_wavefile_ix+wdata.mix_wavefile_offset];

```

Comment:
I think I should keep this as is, as sometimes mix_wavefile_ix is incremented, and  the mix_wavefile_ix+mix_wavefile_offset is used both before and after at times. Surely the compiler would optimize the addition out  where it can?

Reply:
My comment was more about trying to improve the clarity of the code, to make it easier to see where the same value is being used, rather than about optimisation.